src/Meziantou.Analyzer/Rules/AvoidUsingRedundantElseAnalyzer.cs

```diff
@@ -27,30 +27,43 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            context.RegisterSyntaxNodeAction(AnalyzeElseClause, SyntaxKind.ElseClause);
+            context.RegisterSemanticModelAction(AnalyzeSemanticModel);
         }
 
-        private static void AnalyzeElseClause(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
         {
-            var elseClauseSyntax = (ElseClauseSyntax)context.Node;
-            if (elseClauseSyntax is null)
+            var rootNode = context.SemanticModel.SyntaxTree.GetRoot();
+            var elseClauses = rootNode.DescendantNodes().Where(n => n.IsKind(SyntaxKind.ElseClause)).Cast<ElseClauseSyntax>();
+            var locations = elseClauses
+                .Select(clause => AnalyzeElseClause(context, clause))
+                .Where(location => location != null);
+
+            if (!locations.Any())
                 return;
 
+            var diagnostic = Diagnostic.Create(s_rule, locations.First(), locations.Skip(1));
+            context.ReportDiagnostic(diagnostic);
```

Comment:
I don't like the fact that we report only one diagnostic for the whole file because this means a user cannot fix only one of them. Instead, they must fix all of them at once.
So, maybe this is not the right way to solve the issue. However, I think you can create a FixAllProvider that apply the same approach as here. This means:
- Report all diagnostic as before
- Implement a fix all fixer ([documentation](https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md)) that apply the fix in the reverse order (you can sort them by location). 

There are a few examples of Batch Fixers in the Roslyn repository ([example](https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs), [example](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaWarningBatchFixAllProvider.cs), [example](https://github.com/dotnet/roslyn-analyzers/blob/master/src/PublicApiAnalyzers/Core/CodeFixes/DeclarePublicApiFix.cs)). This doesn't think that hard to create. However, we don't to update the code to bw able to trigger the batch fixer when needed.


Another solution is to not support fix all by returning null.

````
public override FixAllProvider GetFixAllProvider() => null;
````

Reply:
I initially considered (albeit briefly) implementing a FixAllProvider. However, I quickly realized that `AvoidUsingRedundantElseFixer.GetFixAllProvider` was never called during testing... thus the proposed fix.

Reply:
I've created a PR to support batch fixer. It's not tested thoroughly but I think you can integrate it in your branch if you want to test the batch fixer

https://github.com/meziantou/Meziantou.Analyzer/pull/150

Reply:
Ok, so based on your batch-fixer branch here's a new implementation.

Reply:
The implementation looks very good! I just add a few minor comments.